### PR TITLE
[feature](Nereids )generate left deep tree when stats is unknown

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

generate left deep tree when stats is unknown

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by jackwener — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java [hunk 1]
```diff
@@ -302,17 +303,53 @@ public Cost visitPhysicalHashJoin(
...
+                    rightRowCount,
+                    0);
+        }
         return CostV1.of(leftRowCount + rightRowCount + outputRowCount,
                 rightRowCount,
                 0
         );
     }
 
+    private boolean isStatsUnknown(PhysicalHashJoin<? extends Plan, ? extends Plan> join,
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:329]: PhysicalHashJoin -> AbstractPhysicalJoin
use AbstractPhysicalJoin, and this function can be used both for hash join and nested loop join

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHashJoin.java [hunk 2]
```diff
@@ -244,6 +246,11 @@ public boolean pushDownRuntimeFilter(CascadesContext context, IdGenerator<Runtim
         return pushedDown;
     }
 
+    public Set<Slot> getConditionSlot() {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHashJoin.java:249]: move to AbstractPhysicalJoin

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalNestedLoopJoin.java [hunk 3]
```diff
@@ -173,6 +176,11 @@ public boolean isBitMapRuntimeFilterConditionsEmpty() {
         return bitMapRuntimeFilterConditions.isEmpty();
     }
 
+    public Set<Slot> getConditionSlot() {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalNestedLoopJoin.java:179]: move to AbstractPhysicalJoin

**Review by englefly — APPROVED** [review]

**Pull request merged**